File: atcoder.jp/contests/abs/tasks/abc088_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];
const print = (...messages: any) => console.log(...messages);

const sort = <T extends number | bigint>(array: T[], reverse = false) =>
  reverse
    ? array.slice().sort((a, b) => (a > b ? -1 : a < b ? 1 : 0))
    : array.slice().sort((a, b) => (a < b ? -1 : a > b ? 1 : 0));

const step2 = <T>(array: T[]) => array.filter((_, i) => i % 2 === 0);

const sum = (array: number[]) => array.reduce((a, b) => a + b, 0);

const tail = <T>(array: T[]) => array.slice(1);

// input functions -------------------------------------------------------------
const inputNumberArray = () => input().split(" ").map(Number);
// input functions -------------------------------------------------------------

const main = () => {
  const N = +input();
  const A = inputNumberArray();

  const sortedA = sort(A, true);
  const alice = sum(step2(sortedA));
  const bob = sum(step2(tail(sortedA)));
  const ans = alice - bob;

  print(ans);
};

main();

File: atcoder.jp/contests/abs/tasks/abc088_b/main.cpp
#define _GLIBCXX_DEBUG
#include <bits/stdc++.h>
using namespace std;

// print functions -------------------------------------------------------------
template <typename T>
void print(T value) {
  cout << value << endl;
}
// print functions -------------------------------------------------------------

template <typename T>
vector<T> sort(const vector<T> &vec, bool reverse = false) {
  vector<T> ret = {vec.begin(), vec.end()};
  if (reverse)
    sort(ret.begin(), ret.end(), greater());
  else
    sort(ret.begin(), ret.end());
  return ret;
}

template <typename T>
vector<T> step2(const vector<T> &vec) {
  vector<T> ret;
  for (size_t i = 0; i < vec.size(); i++)
    if (i % 2 == 0) ret.push_back(vec[i]);
  return ret;
}

template <typename T>
T sum(const vector<T> &vec) {
  return accumulate(vec.begin(), vec.end(), 0.0);
}

template <typename T>
vector<T> tail(const vector<T> &vec) {
  return {vec.begin() + 1, vec.end()};
}

// input functions -------------------------------------------------------------
vector<int> input_vector(int n) {
  vector<int> ret(n);
  for (auto &&i : ret) cin >> i;
  return ret;
}
// input functions -------------------------------------------------------------

int main() {
  int N;
  cin >> N;
  auto A = input_vector(N);

  auto sorted_A = sort(A, true);
  int alice = sum(step2(sorted_A));
  int bob = sum(step2(tail(sorted_A)));
  int ans = alice - bob;

  print(ans);
}
